plugins/cilium-cni/lib/deletion_queue.go

```diff
@@ -164,8 +165,35 @@ func (dc *DeletionFallbackClient) EndpointDeleteMany(req *models.EndpointBatchDe
 // indicating the caller to attempt fallback logic.
 func (dc *DeletionFallbackClient) deleteEndpointsBatch(req *models.EndpointBatchDeleteRequest) (bool, error) {
 	if err := dc.tryConnect(); err != nil {
-		// Failed to setup cilium client.
-		return true, err
+		// Check if agent is starting up. If so, it is about to handle the
+		// deletion queue, thus we should avoid taking the lock in order to
+		// reduce contention.
+		// Instead, we wait for it to complete its bootstrap and retry to
+		// connect again later.
+		var dirExists, socketExists bool
+		if _, err := os.Stat(filepath.Dir(client.DefaultSockPath())); errors.Is(err, fs.ErrNotExist) {
+			dirExists = true
+		}
+		if _, err := os.Stat(client.DefaultSockPath()); errors.Is(err, fs.ErrNotExist) {
+			socketExists = true
+		}
+		if !dirExists || !socketExists {
```

Comment:
Oh, I see that the naming confused me in my comments above. I guess either the variables should be renamed to `*NotExists` or the suggestions above should be applied and the condition below be changed to

```suggestion
		if dirExists && socketExists {
```

Reply:
Good catch. I guess I started with "let's check if the socker/dir exists" in mind and then changed idea later, forgetting about the var names. I'll change them to `*NotExists`. :+1: